docs/source/api/v2/deliveryservices_id_urlkeys.rst

```diff
@@ -0,0 +1,93 @@
+..
+..
+.. Licensed under the Apache License, Version 2.0 (the "License");
+.. you may not use this file except in compliance with the License.
+.. You may obtain a copy of the License at
+..
+..     http://www.apache.org/licenses/LICENSE-2.0
+..
+.. Unless required by applicable law or agreed to in writing, software
+.. distributed under the License is distributed on an "AS IS" BASIS,
+.. WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+.. See the License for the specific language governing permissions and
+.. limitations under the License.
+..
+
+.. _to-api-deliveryservices-id-urlkeys:
+
+***********************************
+``deliveryservices/{{ID}}/urlkeys``
+***********************************
+
+``GET``
+=======
+.. seealso:: :ref:`to-api-deliveryservices-xmlid-xmlid-urlkeys`
+
+Retrieves URL signing keys for a :term:`Delivery Service`.
+
+.. caution:: This method will return the :term:`Delivery Service`'s **PRIVATE** URL signing keys! Be wary of using this endpoint and **NEVER** share the output with anyone who would be unable to see it on their own.
+
+:Auth. Required: Yes
+:Roles Required: None
+:Response Type:  Object
+
+Request Structure
+-----------------
+.. table:: Request Path Parameters
+
+	+------+----------------------------------------------------------------------------------------+
+	| Name | Description                                                                            |
+	+======+========================================================================================+
+	| id   | Filter for the :term:`Delivery Service` identified by this integral, unique identifier |
+	+------+----------------------------------------------------------------------------------------+
+
+.. code-block:: http
+	:caption: Request Example
+
+	GET /api/2.0/deliveryservices/1/urlkeys HTTP/1.1
+	User-Agent: python-requests/2.22.0
+	Accept-Encoding: gzip, deflate
+	Accept: */*
+	Connection: keep-alive
+	Cookie: mojolicious=...
+
+Response Structure
+------------------
+:key<N>: The private URL signing key for this :term:`Delivery Service` as a base-64-encoded string, where ``<N>`` is the "generation" of the key e.g. the first key will always be named ``"key0"``. Up to 16 concurrent generations are retained at any time (``<N>`` is always on the interval [0,16])
```

Comment:
I think this interval is actually `[0,15]` - or, equivalently in the ℤ domain (and we're implicitly in ℕ which is a subset of ℤ), `[0, 16)`.

Did you copy that from another endpoint? It looks familiar, so it might be wrong there, too.

Reply:
> Did you copy that from another endpoint? It looks familiar, so it might be wrong there, too.

Yep, ripped straight from [deliveryservices_xmlid_xmlid_urlkeys.rst](https://github.com/apache/trafficcontrol/blob/93520432ec/docs/source/api/v2/deliveryservices_xmlid_xmlid_urlkeys.rst#response-structure).

Reply:
Changed to `[0, 15]` in cb4e3b56e.